parquet/read.go

```diff
@@ -81,41 +81,42 @@ func reverseTransformArray(f arrow.Field, col arrow.Array) arrow.Array {
 		return reverseTransformMAC(col.(*array.String))
 	case arrow.TypeEqual(dt, types.ExtensionTypes.JSON):
 		return reverseTransformJSON(col.(*array.String))
-	case arrow.TypeEqual(col.DataType(), arrow.FixedWidthTypes.Timestamp_ms):
+	case arrow.TypeEqual(dt, arrow.FixedWidthTypes.Timestamp_ns) || arrow.TypeEqual(dt, arrow.FixedWidthTypes.Timestamp_s):
```

Comment:
timestamps were discussed in the older PR: it somehow persists everything as ms I think... for sure it coerces all seconds to ms (even though timestamp coercion is off by default), but other timestamp units should remain as-is (but they don't?).

Reply:
Yeah exactly. If it's a bug we should prob fix it in arrow I was mostly just trying to get the tests to pass to see where they fail next.